In [1]:
import pandas as pd # data manipulation and analysis
import numpy as np # python lib for working with arrays, linear algebra
import seaborn as sns # data exploration and visualisation
import matplotlib.pyplot as plt # data visulisation and graphical charting
from sklearn.model_selection import train_test_split # create 2 subsets of the data (training and testing)
from sklearn.metrics import accuracy_score # accuracy of correctly classified among all samples.
from sklearn.metrics import classification_report # generating a human-readable text report
import re # determine if a given text fits the given regular expression 
import string 

In [5]:
data_fake = pd.read_csv('Datasets/Fake.csv')
data_true = pd.read_csv('Datasets/True.csv')

data_fake.head()

title  \
0   Donald Trump Sends Out Embarrassing New Year’...   
1   Drunk Bragging Trump Staffer Started Russian ...   
2   Sheriff David Clarke Becomes An Internet Joke...   
3   Trump Is So Obsessed He Even Has Obama’s Name...   
4   Pope Francis Just Called Out Donald Trump Dur...   

                                                text subject  \
0  Donald Trump just couldn t wish all Americans ...    News   
1  House Intelligence Committee Chairman Devin Nu...    News   
2  On Friday, it was revealed that former Milwauk...    News   
3  On Christmas day, Donald Trump announced that ...    News   
4  Pope Francis used his annual Christmas Day mes...    News   

                date  
0  December 31, 2017  
1  December 31, 2017  
2  December 30, 2017  
3  December 29, 2017  
4  December 25, 2017

In [6]:
data_true.head()

title  \
0  As U.S. budget fight looms, Republicans flip t...   
1  U.S. military to accept transgender recruits o...   
2  Senior U.S. Republican senator: 'Let Mr. Muell...   
3  FBI Russia probe helped by Australian diplomat...   
4  Trump wants Postal Service to charge 'much mor...   

                                                text       subject  \
0  WASHINGTON (Reuters) - The head of a conservat...  politicsNews   
1  WASHINGTON (Reuters) - Transgender people will...  politicsNews   
2  WASHINGTON (Reuters) - The special counsel inv...  politicsNews   
3  WASHINGTON (Reuters) - Trump campaign adviser ...  politicsNews   
4  SEATTLE/WASHINGTON (Reuters) - President Donal...  politicsNews   

                 date  
0  December 31, 2017   
1  December 29, 2017   
2  December 31, 2017   
3  December 30, 2017   
4  December 29, 2017

In [7]:
data_fake["class"] = 0
data_true["class"] = 1

In [8]:
data_fake.shape, data_true.shape # returns the shape of an array which is a tuple of integers containing entries 

((23481, 5), (21417, 5))

data_fake contains 23481 rows and 5 columns
data_true contains 21417 rows and 5 columns

In [14]:
print(data_fake.index)
print(data_true.index)

RangeIndex(start=0, stop=23471, step=1)
RangeIndex(start=0, stop=21417, step=1)


In [18]:
data_fake_manual_testing = data_fake.tail(10)
for i in range(23480, 23469, -1):
    if i in data_fake.index:
        data_fake.drop([i], axis = 0, inplace = True)
        
data_true_manual_testing = data_true.tail(10)
for i in range(21416, 21405, -1):
    if i in data_true.index:
        data_true.drop([i], axis = 0, inplace = True) # iterates through the rows and drops 

### train-test splitting

- Evaluating the performance of the ML model on unseen data. By removing a subset of the data we can simulate how the model would perform on new, unseen data.
- The test set is used to evaluate the model's performance by comparing the predicted labels to the actual labels.


In [19]:
data_fake.shape, data_true.shape

((23470, 5), (21406, 5))